python/cuml/common/memory_utils.py

```diff
@@ -37,6 +38,38 @@
         pass
 
 
+class MemoryType(Enum):
```

Comment:
I know I suggested putting this in memory_utils, but I wonder if it'd be more clear to put both enums of device and memory together? @wphicks what do you think?

Reply:
I think I actually prefer it in with memory_utils. I think it helps reinforce the distinction that we're making between device and memory type.